test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -10051,6 +10051,30 @@ BOOST_AUTO_TEST_CASE(delegatecall_return_value)
 	BOOST_CHECK(callContractFunction("get_delegated()") == encodeArgs(u256(1)));
 }
 
+BOOST_AUTO_TEST_CASE(function_types_sig)
+{
+	char const* sourceCode = R"(
+		contract C {
+			function f() returns (bytes4) {
+				return this.f.selector;
+			}
+			function g() returns (bytes4) {
+				function () external returns (bytes4) fun = this.f;
+				return fun.selector;
+			}
+			function h() returns (bytes4) {
+				function () external returns (bytes4) fun = this.f;
```

Comment:
This works now, but not sure we are doing the right thing, since the function signature of this function type can be different depending on what function was assigned to it.

Reply:
Sure, but the correct type signature has to be there, since we also have to know it when we want to call the function. What are your worries?

Reply:
Never mind.